src/java/org/apache/cassandra/db/compaction/UnifiedCompactionContainer.java

```diff
@@ -141,6 +146,16 @@ public CompactionStrategyContainer reload(@Nonnull CompactionStrategyContainer p
         return create(previous, factory, compactionParams, reason, enableAutoCompaction);
     }
 
+    @Override
+    public boolean shouldReload(CompactionParams params, ReloadReason reason, boolean isVectorTable)
+    {
+        if (hasVector() != isVectorTable)
```

Comment:
We only need to check this if the reason is `METADATA_CHANGE`, i.e. I'd put this as `|| hasVector() != isVectorTable` at the end of the `return` line below.